jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -76,19 +79,93 @@ def on_mouse_or_key_event(self, data):
                 self.label_mouseover.value = ""
                 return
 
-            fmt = 'x={:+10.5e} y={:+10.5e}'
-            self.label_mouseover.pixel = fmt.format(x, y)
+            # Snap to the closest data point, not the actual mouse location.
+            sp = None
+            closest_i = None
+            closest_wave = None
+            closest_flux = None
+            closest_label = ''
+            closest_distance = None
+            for lyr in self.state.layers:
+                if not lyr.visible:
+                    continue
+                if isinstance(lyr.layer, GroupedSubset):
+                    if not isinstance(lyr.layer.subset_state, RoiSubsetState):
+                        # then this is a SPECTRAL subset
+                        continue
+                elif ((not isinstance(lyr.layer, BaseData)) or (lyr.layer.ndim not in (1, 3))
+                        or (not lyr.visible)):
+                    continue
+
+                try:
+                    # Cache should have been populated when spectrum was first plotted.
+                    # But if not (maybe user changed statistic), we cache it here too.
+                    statistic = getattr(self.state, 'function', None)
+                    cache_key = (lyr.layer.label, statistic)
+                    if cache_key in self.jdaviz_app._get_object_cache:
+                        sp = self.jdaviz_app._get_object_cache[cache_key]
+                    else:
+                        sp = self.jdaviz_app.get_data_from_viewer('spectrum-viewer',
```

Comment:
Could we change `'spectrum-viewer'` to `self.jdaviz_app._default_spectrum_viewer_reference_name`?

Reply:
Done. Good catch!